### PR TITLE
Remove obsolete example of FleetAllocator service

### LINKED ISSUE #1584: Cleanup: Remove obsolete example of FleetAllocator service
Some parts, for example, waiting for Ready Replicas is redundant in this example code:
https://github.com/googleforgames/agones/commit/cd874a2e1a8427fcb5311dbc0ed686f66ff8bfb4
This code is misleading and should be removed soon.
`GameServerAllocation` should be used for this purpose:
https://agones.dev/site/docs/getting-started/create-fleet/#4-allocate-a-game-server-from-the-fleet
`AllocatorService` could be used also:
https://agones.dev/site/docs/advanced/allocator-service/

### PR DESCRIPTION
**Pull request by akremsa**
**What type of PR is this?**
> /kind cleanup

**What this PR does / Why we need it**:
Removed this part because we don't use it anymore and it leads to some confusions.

**Which issue(s) this PR fixes**:
Closes #1584

### CODE REVIEW

File: site/content/en/docs/Examples/_index.md [hunk 1]
```diff
@@ -35,7 +35,6 @@ These are all examples of simple game server implementations, that integrate the
 
 ## Building on top of Agones
 
-- {{< ghlink href="examples/allocator-service" >}}Allocator Service{{< /ghlink >}} (Go) - 
+- {{< ghlink href="examples/crd-client" >}}Agones API Usage Example{{< /ghlink >}} (Go) -
```
Comment (markmandel) [inline site/content/en/docs/Examples/_index.md:38]: Excellent replacement!

File: site/content/en/docs/Guides/access-api.md [hunk 2]
```diff
@@ -302,4 +302,4 @@ section may also provide the more details on the API conventions that are used i
 
 ## Next Steps
 
-- Learn how to interact with Agones programmatically through the API while creating an [Allocator Service]({{< relref "../Tutorials/allocator-service-go.md" >}}).
+- Learn how to interact with Agones programmatically, see [Allocator Service]({{< relref "../Advanced/allocator-service.md" >}}) tutorial.
```
Comment (markmandel) [inline site/content/en/docs/Guides/access-api.md:305]:
```suggestion
- Learn how to use the [Allocator Service]({{< relref "../Advanced/allocator-service.md" >}}) for single and multi-cluster Allocation.
```

Something like that? WDYT?

File: site/content/en/docs/Getting Started/create-fleet.md [hunk 3]
```diff
@@ -359,4 +359,4 @@ You have now deployed a new version of your game!
 - Have a look at the [GameServer Creation, Allocation and Shutdown Lifecycle]({{< ref "/docs/Guides/gameserver-lifecycle.md" >}}) diagram,
     to give you a good overview of how all the pieces fit together - from creating to integrating with a matchmaker.
 - Or if you want to try to use your own GameServer container make sure you have properly integrated the [Agones SDK]({{< ref "/docs/Guides/Client SDKs/_index.md" >}}).
-- If you would like to learn how to programmatically allocate a Game Server from the fleet using the Agones API, see the [Allocator Service]({{< relref "../Tutorials/allocator-service-go.md" >}}) tutorial.
+- If you would like to learn how to programmatically allocate a Game Server from the fleet using the Agones API, see [Allocator Service]({{< relref "../Advanced/allocator-service.md" >}}) tutorial.
```
Comment (markmandel) [inline site/content/en/docs/Getting Started/create-fleet.md:362]: Maybe point to both https://agones.dev/site/docs/guides/access-api/ and https://agones.dev/site/docs/advanced/allocator-service/ ?

Something like?

```suggestion
- If you would like to learn how to programmatically allocate a Game Server from the fleet, see how to [Access Agones via the Kubernetes API](https://agones.dev/site/docs/guides/access-api/) or alternatively use the [Allocator Service]({{< relref "../Advanced/allocator-service.md" >}}), depending on your needs.
```

File: site/content/en/docs/Guides/access-api.md (hunk 2 above)
Comment (akremsa) [reply site/content/en/docs/Guides/access-api.md:305]: Good suggestion

File: site/content/en/docs/Getting Started/create-fleet.md (hunk 3 above)
Comment (akremsa) [reply site/content/en/docs/Getting Started/create-fleet.md:362]: Agree

**Comment by markmandel** [comment]
Just realised, in this PR we didn't remove: https://agones.dev/site/docs/tutorials/allocator-service-go/

We should probably do that :smile:

**Comment by akremsa** [comment]
@markmandel are you talking about create-fleet.md ? I've put a relref there.

**Pull request merged**